framework/player/init/unitTracker.sqf

```diff
@@ -12,35 +13,39 @@ lmf_player_tracklist = [];
 
 // ADD PLAYERS CYCLE //////////////////////////////////////////////////////////////////////////////
 [] spawn {
-	private _fnc_addUnitMarker =  {
-		private _unit = _this select 0;
-		private _marker = createMarkerLocal ["nk_m_"+str(_unit),getposASL _unit];
-		_marker setMarkerShapeLocal "Icon";
-		_marker setMarkerColorLocal var_markerSide;
-		_marker setMarkerTypeLocal "mil_dot_noshadow";
-		_marker setmarkerSizeLocal [0.4,0.4];
+	private _fnc_addUnit =  {
+		private _unit = _this#0;
 		lmf_player_tracklist pushbackunique _unit;
 	};
+
 	while {true} do {
+		//ADD PLAYERS FROM PLAYER GROUP TO LIST (MINUS GROUP LEADER)
+		{
+			if (_x in lmf_player_tracklist) then {} else {0 =[_x] call _fnc_addUnit};
+		} forEach (units group player - [leader group player]);
+		//REMOVE FROM TRACKLIST IF LEFT GROUP OR BECAME GROUP LEADER OR BECAME OBJNULL
 		{
-			if (_x in lmf_player_tracklist) then {} else {0 =[_x,""] call _fnc_addUnitMarker;};
-			false
-		} count (playableunits + switchableunits - entities "Headless_F");
-		sleep 20;
+			if (group _x != group player || {_x == leader group player || {isNull _x}}) then {
+				lmf_player_tracklist = lmf_player_tracklist - [_x];
+			};
+		} forEach lmf_player_tracklist;
+		sleep 15;
 	};
 };
 
 
 // TRACKING MAGIC /////////////////////////////////////////////////////////////////////////////////
-while {true} do {
-	waitUntil {count lmf_player_tracklist > 0};
+private _trackingEH = ((findDisplay 12) displayCtrl 51) ctrlAddEventHandler ["Draw",{
+	if (!visibleMap || {lmf_player_tracklist isEqualTo []}) exitWith {};
 	{
-		private _marker = ("nk_m_"+ str(_x));
-		_marker setmarkerposlocal getposASL _x;
-		if (isNull _x) then {
-			lmf_player_tracklist = lmf_player_tracklist - [_x];
-			deleteMarkerLocal _marker;
+		private _color = [assignedTeam _x] call {
+			if (_this#0 == "RED") exitwith {[1,0,0,1]};
+			if (_this#0 == "GREEN") exitwith {[0,1,0,1]};
+			if (_this#0 == "BLUE") exitwith {[0,0,1,1]};
+			if (_this#0 == "YELLOW") exitwith {[1,1,0,1]};
+			[1,1,1,1]
 		};
+		(_this#0) drawIcon ["iconMan",[0,0,0,1],getPosASL _x,20,20,getDir _x];
```

Comment:
If you need to use a `select` statement more then once, might consider of loading what you need into a private variable. In this case you are doing `_this#0`for ever unit tracked, while you only need to assign it once outside the `count`